com.unity.ml-agents.extensions/Runtime/Match3/Match3Actuator.cs

```diff
@@ -62,40 +58,47 @@ public void OnActionReceived(ActionBuffers actions)
             }
             var moveIndex = actions.DiscreteActions[0];
 
-            if (m_Board.Rows != m_Rows || m_Board.Columns != m_Columns || m_Board.NumCellTypes != m_NumCellTypes)
             {
-                Debug.LogWarning(
-                    $"Board shape changes since actuator initialization. This may cause unexpected results. " +
-                    $"Old shape: Rows={m_Rows} Columns={m_Columns}, NumCellTypes={m_NumCellTypes} " +
-                    $"Current shape: Rows={m_Board.Rows} Columns={m_Board.Columns}, NumCellTypes={m_Board.NumCellTypes}"
-                );
+                var currentMaxBoardSize = m_Board.GetMaxBoardSize();
+
+                if (!currentMaxBoardSize.Equals(m_MaxBoardSize))
```

Comment:
TODO: change this check to make sure GetCurrentBoardSize < m_MaxBoardSize (for each field)